api/v1beta1/module_webhook.go

```diff
@@ -95,19 +95,37 @@ func (m *Module) validateKernelMapping() error {
 
 func (m *Module) validateModprobe() error {
 	modprobe := m.Spec.ModuleLoader.Container.Modprobe
-	moduleNameDefined := modprobe.ModuleName != ""
+	moduleName := modprobe.ModuleName
+	moduleNameDefined := moduleName != ""
 	rawLoadArgsDefined := modprobe.RawArgs != nil && len(modprobe.RawArgs.Load) > 0
 	rawUnloadArgsDefined := modprobe.RawArgs != nil && len(modprobe.RawArgs.Unload) > 0
 
 	if moduleNameDefined {
 		if rawLoadArgsDefined || rawUnloadArgsDefined {
 			return errors.New("rawArgs cannot be set when moduleName is set")
 		}
-		return nil
+	} else if !rawLoadArgsDefined || !rawUnloadArgsDefined {
+		return errors.New("load and unload rawArgs must be set when moduleName is unset")
 	}
 
-	if !rawLoadArgsDefined || !rawUnloadArgsDefined {
-		return errors.New("load and unload rawArgs must be set when moduleName is unset")
+	if modprobe.ModulesLoadingOrder != nil {
+		if len(modprobe.ModulesLoadingOrder) < 2 {
+			return errors.New("if a loading order is defined, at least two values must be defined")
+		}
+
+		if !moduleNameDefined {
```

Comment:
is there any sense to have a Module without moduleName defined? I think that verification can be moved to line 99

Reply:
in case user uses raw and modules order list, then we need to make sure that the first module on the raw command is also the first on the list. Maybe simplier just to block the modules order list in case user uses raw?

Reply:
I would in fact not check anything related to the module order list if raw args are used.